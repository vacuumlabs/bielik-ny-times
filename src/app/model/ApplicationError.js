import { get } from 'lodash'
// helper functions to parse errors

const makeError = message => ({ message })

export const isApplicationError = error => !!error.message

export function parseArticleError(error) {
  if (!error) {
    return null
  }
  const nytRequestError = get(error, 'fault.faultstring')
  if (nytRequestError) {
    // e.g. invalid api key
    return makeError(nytRequestError)
  }
  const nytFormattedError = get(error, 'errors[0]')
  if (nytFormattedError) {
    // e.g. invalid request params
    return makeError(nytFormattedError)
  }
  return makeError('Unexpected error has occured :(')
}
