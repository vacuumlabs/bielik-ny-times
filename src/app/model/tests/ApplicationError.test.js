import { parseArticleError } from '../ApplicationError'

describe('ApplicationError', () => {
  it('should not return if missing', () =>
    expect(parseArticleError(undefined)).toMatchSnapshot())

  it('should return NYT request error', () =>
    expect(
      parseArticleError({ fault: { faultstring: 'hello error' } })
    ).toMatchSnapshot())

  it('should return NYT formatted error', () =>
    expect(
      parseArticleError({ errors: ['hello error', 'other'] })
    ).toMatchSnapshot())

  it('should return unknown error', () =>
    expect(parseArticleError({ msg: 'something' })).toMatchSnapshot())
})
