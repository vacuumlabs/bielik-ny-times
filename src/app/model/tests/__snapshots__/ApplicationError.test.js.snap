// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationError should not return if missing 1`] = `null`;

exports[`ApplicationError should return NYT formatted error 1`] = `
Object {
  "message": "hello error",
}
`;

exports[`ApplicationError should return NYT request error 1`] = `
Object {
  "message": "hello error",
}
`;

exports[`ApplicationError should return unknown error 1`] = `
Object {
  "message": "Unexpected error has occured :(",
}
`;
